tests/util/localregistry/cmd/localregistry.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/mixer/cmd/shared"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	// Default values for test env setup
+	LocalRegistryFile      = "tests/util/localregistry/localregistry.yaml"
+	LocalRegistryNamespace = "kube-system"
+	LocalRegistryPort      = 5000
+)
+
+type localRegistrySetupArgs struct {
+	remoteRegistryPort uint16
+	kubeconfig         string
+}
+
+func setupCfgCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
```

Comment:
I think we should move the Cobra-specific functions over to the cmd file.  The non-Cobra code can move up to `tests/util/localregistry/localregistry.go`.

Reply:
Do we even need Cobra ?

Reply:
@nmittler : will do that.. 
@costinm : If a user is using environment other than minikube, gke for which we will provide this support by default.. they can still setup in cluster local registry using localregistry cobra command..

Reply:
I think I'm with @costinm in removing the Cobra abstraction ... it's added complexity that isn't always necessary. I actually just did something similar to remove cobra from our framework (#7887).  In my case, Cobra's parse function behaved in a way that made it impossible to run the application from the GoLand IDE.